Task 5 done

AbramonRomka opened the pull request.

AbramonRomka committed:
Task 5 done

ArthurKharatian commented on src/itmo/java/basic/lesson5/t2.java:
```diff
@@ -0,0 +1,29 @@
+    public static boolean reverse (String wordIn){
+        StringBuilder wordBuilder = new StringBuilder(wordIn);
+        String wordRev = String.valueOf(wordBuilder.reverse());
+        return wordIn.equals(wordRev);
```
а если слово Топот?

ArthurKharatian commented on src/itmo/java/basic/lesson5/t2.java:
```diff
@@ -0,0 +1,29 @@
+        System.out.println(word);
+        print(reverse(word), word);
+    }
+    public static void print (boolean isPalindrome, String wordOut){
```
для чего этот метод?)

ArthurKharatian commented on src/itmo/java/basic/lesson5/t3.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    public static void refactor(String s) {
+        String[] arrayText = s.split("(?=[А-Я])");
```
метод должен принимать в качестве параметра и строку которую ищем и на которую меняем.

ArthurKharatian commented on src/itmo/java/basic/lesson5/t3.java:
```diff
@@ -0,0 +1,41 @@
+        for (int i = 0; i < arrayText.length; i++) {
+            String[] arrayWord = arrayText[i].split(" ");
+            for (int j = 0; j < arrayWord.length; j++) {
+                if (arrayWord[j].startsWith("бяк")) {
```
а если слово в середине слова? или в конце

ArthurKharatian commented on src/itmo/java/basic/lesson5/t5.java:
```diff
@@ -0,0 +1,24 @@
+        System.out.print("Строка после инверсии: ");
+        System.out.println(inverted(str));
+    }
+}
```
во всех задания методы должны возвращать результат их выполнения. в мейн вы можете эти результаты выводить на экран. и через него же передавать данные на проверку

ArthurKharatian replied:
+ нейминг методов виде print неинформативно. название метода должно нам говорить что данный метод выполняет максимально коротко (без крайностей, типа аббревиатур)

ArthurKharatian commented on src/itmo/java/basic/lesson5/t1.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    public static String trMethod(String lt) {
+        String[] arrayText = lt.split("[., ()]");
```
лучше сразу очистить от знаков пунктуации

AbramonRomka committed:
Task 5 fix:
1. removed print method and add function remove punctuation marks and fix condition if.
2. remove print method and add toLowerCase function.
3. remove print method and add passing a string in refactor method and .startWith replaice by .contains.
4. remove print method.
5. remove print method.

ArthurKharatian commented on src/itmo/java/basic/lesson5/t2.java:
```diff
@@ -0,0 +1,24 @@
+    public static boolean reverse (String wordIn, String word){
+        StringBuilder wordBuilder = new StringBuilder(wordIn);
+        String wordRev = String.valueOf(wordBuilder.reverse());
+        return wordIn.equals(wordRev);
```
слово Топот так и не пройдет

ArthurKharatian commented on src/itmo/java/basic/lesson5/t3.java:
```diff
@@ -0,0 +1,34 @@
+        System.out.println("");
+    }
+
+    public static String[] refactor(String Word, String refactorWord) {
```
тут должны быть в качестве параметров исходная строка, строка которую ищем и строка на которую заменяем. метод должен вернуть строку с изменениями, не массив

ArthurKharatian commented on src/itmo/java/basic/lesson5/t3.java:
```diff
@@ -0,0 +1,34 @@
+        for (int i = 0; i < arrayText.length; i++) {
+            String[] arrayWord = arrayText[i].split(" ");
+            for (int j = 0; j < arrayWord.length; j++) {
+                if (arrayWord[j].contains("бяк")) {
```
тут вы хардкодите, а надо проверять текст на содержание переданного слова

AbramonRomka committed:
Task 5 v.2.0. Global modernise =)

ArthurKharatian commented on src/itmo/java/basic/lesson5/t2.java:
```diff
@@ -0,0 +1,23 @@
+    public static boolean reverse(String word) {
+        StringBuilder wordBuilder = new StringBuilder(word);
+        String wordRev = String.valueOf(wordBuilder.reverse());
+        return word.equals(wordRev);
```
регистр не нужно учитывать

AbramonRomka committed:
Update t2.java
